dpnp/backend/kernels/dpnp_krnl_indexing.cpp

```diff
@@ -175,6 +175,25 @@ void dpnp_fill_diagonal_c(void* array1_in, void* val_in, size_t* shape, const si
     return;
 }
 
+template<typename _DataType, typename _MaskType>
```

Comment:
it look like _maskType is always one type. Maybe it is not nessesary to put this as a typeneme here and just fix type of input `mask_in` as boolean or integer